lib/auth_config.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+class AuthConfig
+
+  AUTH_CONFIG_PATH = Rails.root.join('config/auth.yml')
+
+  class << self
+    def client_id
+      get_var_from_environment(:client_id)
+    end
+
+    def secret
+      get_var_from_environment(:secret)
+    end
+
+    def host_url
+      get_var_from_environment(:host_url)
+    end
+
+    def realm
+      get_var_from_environment(:realm)
+    end
+
+    def admin_role
+      get_var_from_environment(:admin_role, required: false)
+    end
+
+    private
+
+    def get_var_from_environment(key, required: true)
+      if !locale? && required
+        ENV[key.to_s] || raise("Environment variable not set: '#{key}'")
+      elsif !locale?
```

Comment:
`unless`

Reply:
If i use unless then I'm not able to use elseif. 
Additionally there isn't a huge diffrence beween 
`if !locale? && required` and `unless locale? && !required` since in both cases there is one negated param